src/overlays/actors/ovl_Bg_Dy_Yoseizo/z_bg_dy_yoseizo.c

```diff
@@ -38,56 +37,730 @@ ActorInit Bg_Dy_Yoseizo_InitVars = {
+                scale = 1.0f;
+                life = 90;
+
+                velocity.x = accel.x * 100.0f;
+                velocity.y = accel.y * 100.0f;
+                velocity.z = accel.z * 100.0f;
+
+                effectType = GREAT_FAIRY_GET_TYPE(&this->actor);
+
+                pos.x = this->actor.world.pos.x;
+                pos.y = this->actor.world.pos.y + spawnHeightVariation +
+                        (Rand_ZeroOne() - 0.5f) * (spawnHeightVariation * 0.5f);
+                pos.z = this->actor.world.pos.z + 30.0f;
+                break;
+
+            case GREAT_FAIRY_EFFECT_TRAJECTORY_RADIANT:
+                scale = 1.0f;
+                life = 90;
+
+                effectType = GREAT_FAIRY_GET_TYPE(&this->actor);
+
+                pos.x = this->actor.world.pos.x;
+                pos.y = this->actor.world.pos.y + spawnHeightVariation +
+                        (Rand_ZeroOne() - 0.5f) * (spawnHeightVariation * 0.5f);
+                pos.z = this->actor.world.pos.z + 30.0f;
+                break;
+
+            default: // all become convergent type
+                scale = 0.2f;
+                life = 50;
+
+                effectType = GREAT_FAIRY_EFFECT_TRAJECTORY_CONVERGE_ON_PLAYER;
+
+                pos.x = Rand_CenteredFloat(10.0f) + this->actor.world.pos.x;
+                pos.y = this->actor.world.pos.y + spawnHeightVariation + 50.0f +
+                        (Rand_ZeroOne() - 0.5f) * (spawnHeightVariation * 0.1f);
+                pos.z = this->actor.world.pos.z + 30.0f;
+                break;
+        }
+
+        primColor.r = sEffectPrimColors[effectType].r;
+        primColor.g = sEffectPrimColors[effectType].g;
+        primColor.b = sEffectPrimColors[effectType].b;
+        envColor.r = sEffectEnvColors[effectType].r;
+        envColor.g = sEffectEnvColors[effectType].g;
+        envColor.b = sEffectEnvColors[effectType].b;
+
+        BgDyYoseizo_SpawnEffect(this, &pos, &velocity, &accel, &primColor, &envColor, scale, life, effectType);
+    }
+}
+
+void func_80A0AD50(BgDyYoseizo* this) {
+    f32 scale = this->actor.scale.x;
+    f32 heightTarget = this->actor.home.pos.y + 40.0f;
+
+    Math_ApproachF(&this->actor.world.pos.y, heightTarget, this->unk2F0, 100.0f);
+    Math_ApproachF(&scale, 0.035f, this->unk2F4, 0.005f);
+    Math_ApproachF(&this->unk2F0, 0.8f, 0.1f, 0.02f);
+    Math_ApproachF(&this->unk2F4, 0.2f, 0.03f, 0.05f);
+    BgDyYoseizo_SpawnEffects(this, GREAT_FAIRY_EFFECT_TRAJECTORY_RADIANT, 2);
+    Actor_SetScale(&this->actor, scale);
+}
+
+void func_80A0AE1C(BgDyYoseizo* this, PlayState* play) {
+    f32 scale = this->actor.scale.x;
+    f32 heightTarget = this->actor.home.pos.y;
+
+    if (scale < 0.003f) {
+        this->actionFunc = func_80A0BB08;
+        Actor_SetScale(&this->actor, 0.0f);
+        this->unk2F0 = 0.0f;
+        this->unk2F4 = 0.0f;
+        this->unk2F8 = 0;
+
+        if (GREAT_FAIRY_GET_TYPE(&this->actor) < 4) {
```

Comment:
```suggestion
        if (GREAT_FAIRY_GET_TYPE(&this->actor) <= GREAT_FAIRY_TYPE_COURAGE) {
```
To avoid hard constants?